File: src/main/java/org/snipsniper/sceditor/ezmode/EzModeStampTab.java
package org.snipsniper.sceditor.ezmode;

import org.snipsniper.sceditor.SCEditorWindow;

import javax.swing.*;
import java.awt.*;
import java.awt.event.MouseAdapter;
import java.awt.event.MouseEvent;
import java.awt.event.MouseMotionAdapter;
import java.awt.image.BufferedImage;

public class EzModeStampTab extends JPanel {
    private final BufferedImage image;

    public EzModeStampTab(BufferedImage image, int size, SCEditorWindow scEditorWindow, int stampIndex) {
        this.image = image;
        setPreferredSize(new Dimension(size, size));
        addMouseListener(new MouseAdapter() {
            @Override
            public void mouseClicked(MouseEvent mouseEvent) {
                scEditorWindow.setSelectedStamp(stampIndex);
            }
        });

        addMouseMotionListener(new MouseMotionAdapter() {
            @Override
            public void mouseDragged(MouseEvent e) {
                scEditorWindow.setSelectedStamp(stampIndex);
            }
        });
    }

    @Override
    public void paint(Graphics g) {
        g.drawImage(image, 0, 0, getWidth(), getHeight(), null);
    }

}

File: src/main/kotlin/net/snipsniper/sceditor/ezmode/EzModeStampTab.kt
package net.snipsniper.sceditor.ezmode

import net.snipsniper.sceditor.SCEditorWindow
import java.awt.Dimension
import java.awt.Graphics
import java.awt.event.MouseAdapter
import java.awt.event.MouseEvent
import java.awt.event.MouseMotionAdapter
import java.awt.image.BufferedImage
import javax.swing.JPanel

class EzModeStampTab(private val image: BufferedImage, size: Int, scEditorWindow: SCEditorWindow, stampIndex: Int): JPanel() {
    init {
        preferredSize = Dimension(size, size)
        addMouseListener(object: MouseAdapter() {
            override fun mouseClicked(e: MouseEvent?) {
                scEditorWindow.setSelectedStamp(stampIndex)
            }
        })
        addMouseMotionListener(object: MouseMotionAdapter() {
            override fun mouseDragged(e: MouseEvent?) {
                scEditorWindow.setSelectedStamp(stampIndex)
            }
        })
    }

    override fun paint(g: Graphics) {
        g.drawImage(image, 0, 0, width, height, null)
    }
}